packages/dts-generator/lib/phases/ast-transform.js

```diff
@@ -215,6 +216,47 @@ function addDefineArrayInterface(ast, symbolTable) {
   }
 }
 
+/**
+ * Adds generic type to `extend` method and modify `oClassInfo` type.
+ * This is done to specify `this` type inside the `oClassInfo` methods.
+ *
+ * @param symbolTable
+ */
+function modifyExtendMethods(symbolTable) {
+  _.forEach(symbolTable, (symbol) => {
+    if (symbol.kind !== "Class") {
+      return;
+    }
+
+    const extendMethod = _.find(
+      symbol.methods,
+      (method) => method.name === "extend"
+    );
+    if (extendMethod === undefined) {
+      return;
+    }
+
+    const oClassInfoParam = _.find(
+      extendMethod.parameters,
+      (param) => param.name === "oClassInfo"
+    );
+    if (oClassInfoParam === undefined) {
+      return;
+    }
+
+    extendMethod.genericType = {
+      kind: "SimpleType",
+      type: "T",
```

Comment:
@tal-sapan proposed that `T` should instead be: `extend<T extends Record<string, unknown>>`

This is mean to **enforce** users to specify an object literal type, which is a constraint that has been lost with this change (previously oClassInfo was of `object` type).

can you evaluate this suggestion? 
- do all scenarios work with it?
- Can any object be passed to .extend ? (e.g even one defined in a separate variable?)
- Would doing `.extend('MyClass', 666) now cause an error?